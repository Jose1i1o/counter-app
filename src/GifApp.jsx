import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from './components/GifGrid';

const GifApp = () => {

  const [ categories, setCategories ] = useState(['X-men']);


  return (
    <>
      <h2>
      🎁 GIFs App 🎁
      </h2>
      <AddCategory setCategories={ setCategories }/>
      <hr />
        {
        categories.map( category =>
          <GifGrid
            key={ category }
            category={ category }/>
        )
        }
    </>
)}

export default GifApp;
