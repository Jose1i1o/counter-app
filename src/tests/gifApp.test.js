// Enzyme not available for React 18. Tested with React 17

import React from 'react';
import { shallow } from 'enzyme';
import GifApp from '../GifApp';

describe('Testing <GifApp />', () => {
    
    test('should display correctly', () => {

        const wrapper = shallow(<GifApp />);
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('should show a list of categories', () => {

        const categories = ['One Punch', 'X-men'];
        const wrapper = shallow(<GifApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
})
