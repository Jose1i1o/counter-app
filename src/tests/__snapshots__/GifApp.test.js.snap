// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifApp /> to display a list of categories 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="X-men"
      key="X-men"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Testing <GifApp /> to display correctly 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
